### PR TITLE
Integration tests

### LINKED ISSUE #3: Add tests to show users how to test snaps
Right now we don't have any tests in this repository. To help users to get an understanding of how to test snaps, we should add some basic tests, at least for the snap.

I'm not sure if tests for the frontend are necessary, but we could add some anyway, to make it easier for users.

### PR DESCRIPTION
**Pull request by mpetrunic**
Closes #3.

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - run: yarn install --immutable
       - run: yarn build
       - run: yarn lint
-      - run: yarn test
+      - run: xvfb-run --auto-servernum  yarn test
```
Comment (ritave) [inline .github/workflows/build-test.yml:35]: @bowensanders Is this how you're running other tests?

File: package.json [hunk 2]
```diff
@@ -23,13 +23,14 @@
     "lint:fix": "yarn lint:eslint --fix && yarn lint:misc --write",
     "lint:misc": "prettier '**/*.json' '**/*.md' '!**/CHANGELOG.md' '**/*.yml' --ignore-path .gitignore",
     "start": "yarn workspaces foreach --parallel --interlaced --verbose run start",
-    "test": "echo \"TODO\""
+    "test": "yarn workspaces foreach run test"
```
Comment (ritave) [inline package.json:26]:
```suggestion
    "test": "yarn workspaces foreach --parallel run test"
```

**Review by ritave — CHANGES_REQUESTED** [review]
@mpetrunic Hey man! Welcome to the MetaMask repos. Could you please use configs from the [Module Template](https://github.com/MetaMask/metamask-module-template)? Especially `jest.config.js`

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (bowensanders) [reply .github/workflows/build-test.yml:35]: In metamask-extension, tests are run using `/test/e2e/run-all.js` - which calls `run-e2e-test.js`
that runs `mocha --no-config --timeout 60000 xxx` where xxx is a test in the test/e2e/snaps folder.
I guess this means that we use mocha to run selenium to run individual tests, and in this case they run (locally) in series, not parallel.

**Comment by mpetrunic** [comment]
> @mpetrunic Hey man! Welcome to the MetaMask repos. Could you please use configs from the [Module Template](https://github.com/MetaMask/metamask-module-template)? Especially `jest.config.js`

Yeah sorry about that, not using jest day-to-day so I just copied some config from your repos.
Also, it seems like Socket security doesn't really understand optional peer dependencies?

**Comment by ritave** [comment]
Don't worry about Socket, I'll make Socket happy when you're ready to merge.

**Comment by mpetrunic** [comment]
> @mpetrunic is this PR still relevant to anything?

It is and it was ready long time ago. Let me see if it needs some update since we removed the need for xvfb?

**Comment by kenhkan** [comment]
Closing this as this has become irrelevant.

**Comment by mpetrunic** [comment]
> Closing this as this has become irrelevant.

How is this irrelevant since issue #3 is still open and this was suppose to resolve it?

**Comment by Mrtenz** [comment]
Hi @mpetrunic, can you please rebase to the latest `main`? That should resolve the issue with the `require-additional-reviewer` workflow.

File: packages/snap/test/integration/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      connectedPage,
+      snapId,
+      'hello',
+    );
+    await metaMask.snaps.dialog.accept();
+    const result = await resultPromise;
+    expect(result).toBeTruthy();
+  });
+
+  test('snap invoke should be true if dialog accepted', async function () {
```
Comment (ziad-saab) [inline packages/snap/test/integration/index.spec.ts:43]:
```suggestion
  test('snap invoke should be false if dialog rejected', async function () {
```

**Pull request closed without merging**